### DISCUSSION TITLE
PNG resolution and SVG rendering

### DISCUSSION THREAD

**Discussion by queezz** [csgoh/roadmapper#68] [Q&A] [answered]
Is there a way to specify PNG dpi, like in matplotlib? The text and the whole diagram is pixilated.

I tried to save in SVG, but the text is out of alignment, and some lines are missing.

**Comment by csgoh** [comment]
@queezz Can you provide me your sample code for me to reproduce the issue?

**Reply by queezz** [reply]
I have to change the code before I can share, I'll get back to you as soon as I can, probably during a week.

**Reply by queezz** [reply]
Hello, this is my code. Nothing special.


```python
from roadmapper.roadmap import Roadmap
from roadmapper.timelinemode import TimelineMode

ext = 'png'
ext = 'svg'
my_roadmap= Roadmap(width=1200, height=800,
                    auto_height=True,
                    show_marker=False, 
                    painter_type=ext,
                    #colour_theme="roadsettings.json",
                    #colour_theme="BLUEMOUNTAIN"
                    colour_theme = "GREENTURTLE"
                    )
my_roadmap.set_title("Plan")
my_roadmap.set_timeline(
    mode=TimelineMode.MONTHLY, start="2023-10-12", number_of_items=5,
     year_font_size=18, 

    )

color = "#FFC000"
group = my_roadmap.add_group("Group 1", fill_colour=color, font_colour="black")
task = group.add_task("Task A", "2023-10-12", "2023-10-30",fill_colour=color, font_colour="black")
task.add_milestone("", "2023-11-15",fill_colour=color)

color = "#ED7D31"
group = my_roadmap.add_group("Group 2", fill_colour=color, font_colour="black")
task = group.add_task("Task A", "2023-11-01", "2023-12-5",fill_colour=color, font_colour="black")

color = "#1dbef0"
group = my_roadmap.add_group("Group 2", fill_colour=color, font_colour="black")
task = group.add_task("Task A", "2023-11-01", "2023-12-5",fill_colour=color, font_colour="black")
task = group.add_task("Task B", "2023-11-01", "2023-12-5",fill_colour=color, font_colour="black")


my_roadmap.draw()
my_roadmap.save(f"problem.{ext}")
```
**PNG** If zoomed in just a little, pixelation is quite obvious. It would be nice to have ability to set dpi of an image. 
![problem](https://github.com/csgoh/roadmapper/assets/12447676/539f1a0d-0d77-4e51-8804-3ad708d6e7c3)

**SVG** Grid lines are missing, text is not centered vertically.
![problem](https://github.com/csgoh/roadmapper/assets/12447676/9b008793-83b0-423b-8fe2-5a8b41712c07)

**Reply by csgoh** [reply, marked as answer]
@queezz I have fixed the SVG issue where vertical lines are missing. You can download v.1.3.3.

As for the PNG painter type, due to how roadmapper is implemented currently, increasing the image dpi does not eliminate the pixelation. I would need more time on this and hope to change how roadmaps are render in the future releases.

**Reply by queezz** [reply]
Thank you for your hard work, appreciate it.
